src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -1330,9 +1294,14 @@ protected override NamedPipeClientStream DoConnect(int timeout)
             }
             catch (Exception)
             {
-                pipeHandle.Dispose();
+                sf?.DangerousRelease();
+                pipeHandle?.Dispose();
                 throw;
             }
+            finally
+            {
+                sf?.Dispose();
```

Comment:
We don't want to dispose `sf` here as I assume it makes the returned pipe stream object invalid.

Reply:
sf.DangerousAddRef() protects from this.